### PR TITLE
Fixing collection recreate issue for changefeed in batch mode

### PR DESCRIPTION
**Pull request by FabianMeiswinkel**
# Description
Fixing collection recreate issue for changefeed in batch mode that is caused by change feed query API (pull mode) using incorrect key for collection cache - as a result it is using the old collection.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/SparkBridgeInternal.scala [hunk 1]
```diff
@@ -67,4 +66,17 @@ private[cosmos] object SparkBridgeInternal {
...
+    val clientWrapper = container.getDatabase.getDocClientWrapper
+
+    val link = container.getLinkWithoutTrailingSlash;
+
+    val obsoleteValue = new DocumentCollection
+    ModelBridgeInternal.setResourceId(obsoleteValue, obsoleteRid)
+
+    clientWrapper
+      .getCollectionCache()
+      .resolveByNameAsync(null, link, null, obsoleteValue)
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/SparkBridgeInternal.scala:80]: should the return type be Mono[Unit]?

File: sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/implementation/SparkBridgeImplementationInternal.scala [hunk 2]
```diff
@@ -269,7 +274,7 @@ private[cosmos] object SparkBridgeImplementationInternal extends BasicLoggingTra
     val continuations = tokens
       .map(token => {
         val pkRangeId = token._1
-        val lsn: Long = token._2
+        val lsn: Long = token._2 + 1
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/implementation/SparkBridgeImplementationInternal.scala:277]: why increase the lsn by 1 here?

File: sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/ChangeFeedBatch.scala [hunk 3]
```diff
@@ -86,10 +87,21 @@ private class ChangeFeedBatch
           if (offsetIsValid) {
             changeFeedStateBase64
           } else {
+            SparkBridgeInternal.clearCollectionCache(
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/ChangeFeedBatch.scala:90]: no subscribe()?

Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/ChangeFeedBatch.scala:90]: Fixed (blocking in the BridgeInternal)

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncContainer.java [hunk 4]
```diff
@@ -758,6 +758,14 @@ <T> CosmosPagedFlux<T> queryChangeFeedInternal(
             queryChangeFeedInternalFunc(cosmosChangeFeedRequestOptions, classType));
     }
 
+    String getLinkWithoutTrailingSlash() {
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncContainer.java:761]: Probably can use Utils.trimBeginningAndEndingSlashes

File: sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/SparkBridgeInternal.scala (hunk 1 above)
Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/SparkBridgeInternal.scala:80]: Fixed - blocking here now.

**Review by xinlian12 — APPROVED** [review]
LGTM, thanks

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncContainer.java (hunk 4 above)
Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncContainer.java:761]: Probably - but not worth risking any regression after ADF tested this. Won't fix.

File: sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/implementation/SparkBridgeImplementationInternal.scala (hunk 2 above)
Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/implementation/SparkBridgeImplementationInternal.scala:277]: Added comments - but yes Spark 2 tracks next-to-be-sent continuation - Spark 3 the LSN of the last processed docs - so offset by 1

**Pull request merged**